include/solana.hpp

```diff
@@ -275,20 +404,60 @@ class Connection {
                                   const size_t length = 0);
   json getBlockhashRequest(const std::string &commitment = "finalized",
                            const std::string &method = "getRecentBlockhash");
-  json sendTransactionRequest(
-      const std::string &transaction, const std::string &encoding = "base58",
-      bool skipPreflight = false,
-      const std::string &preflightCommitment = "finalized");
+
+  /*
+   * send rpc request
+   * @return result from response
+   */
+  json sendJsonRpcRequest(const json &body) const;
+
   ///
   /// 2. Invoke RPC endpoints
   ///
+
+  /**
+   * Request an allocation of lamports to the specified address
+   */
+  std::string requestAirdrop(const PublicKey &pubkey, uint64_t lamports);
+
+  /**
+   * Sign and send a transaction
+   */
+  std::string sendTransaction(
+      const Keypair &keypair, const CompiledTransaction &tx,
+      const SendTransactionConfig &config = SendTransactionConfig()) const;
+
+  /**
+   * Send a transaction that has already been signed and serialized into the
+   * wire format
+   */
+  std::string sendRawTransaction(
+      const std::string &transaction,
+      const SendTransactionConfig &config = SendTransactionConfig()) const;
+
+  /**
+   * Send a transaction that has already been signed, serialized into the
+   * wire format, and encoded as a base64 string
+   */
+  std::string sendEncodedTransaction(
+      const std::string &transaction,
+      const SendTransactionConfig &config = SendTransactionConfig()) const;
+
+  /**
+   * Simulate sending a transaction
+   */
+  json simulateTransaction(const Keypair &keypair,
```

Comment:
please return all data the json rpc api provides: [SimulatedTransactionResponse](https://solana-labs.github.io/solana-web3.js/modules.html#SimulatedTransactionResponse)